src/controllers/productController.ts

```diff
@@ -3,7 +3,19 @@ import {Request, Response, NextFunction} from 'express';
 
 async function retrieveProduct(req: Request, res: Response, next: NextFunction) {
   try {
-    const product = await Product.findByPk(req.params.id);
+    const product = await Product.findByPk(req.params.id, {
```

Comment:
Do the same for here also so we only have business logic in the controllers

Reply:
wdym by business logic

Reply:
Its like the core logic of the application like when a product is created -> need create the ratings and setup analytics for the product etc